### PR TITLE
No temp nuget.config when NuGet 4.8 or higher

### PR DESCRIPTION
**Pull request by satbai**

### CODE REVIEW

File: Tasks/Common/nuget-task-common/NuGetToolRunner2.ts [hunk 1]
```diff
@@ -30,6 +37,8 @@ function prepareNuGetExeEnvironment(
     let env: EnvironmentDictionary = {};
     let originalCredProviderPath: string = null;
     let envVarCredProviderPathV2: string = null;
+    let prefix: string = null;
```
Comment (satbai) [inline Tasks/Common/nuget-task-common/NuGetToolRunner2.ts]: remove the prefix here and the usage beloq

File: Tasks/Common/nuget-task-common/NuGetToolRunner2.ts [hunk 2]
```diff
@@ -90,6 +110,10 @@ function prepareNuGetExeEnvironment(
         }
     }
 
+    if (noTempConfigFile != null) {
```
Comment (satbai) [inline Tasks/Common/nuget-task-common/NuGetToolRunner2.ts]: remove

File: Tasks/Common/nuget-task-common/NuGetToolRunner2.ts [hunk 3]
```diff
@@ -30,6 +37,7 @@ function prepareNuGetExeEnvironment(
     let env: EnvironmentDictionary = {};
     let originalCredProviderPath: string = null;
     let envVarCredProviderPathV2: string = null;
+    let noTempConfigFile: string = null;
```
Comment (satbai) [inline Tasks/Common/nuget-task-common/NuGetToolRunner2.ts]: remove

File: Tasks/Common/nuget-task-common/NuGetToolRunner2.ts [hunk 4]
```diff
@@ -295,3 +308,49 @@ export function getNuGetProxyFromEnvironment(): string {
...
+                    tl.debug(authInfo.packageSource.feedUri);
+                    break;
+                default:
+                    break;
+            }
+        });
+
+        const externalCredentials: string = JSON.stringify(enpointCredentialsJson);
+        return externalCredentials;
+    }
```
Comment (satbai) [inline Tasks/Common/nuget-task-common/NuGetToolRunner2.ts]: return null if not` authInfo && authInfo.externalAuthInfo`

File: Tasks/NuGetCommandV2/nugetrestore.ts [hunk 5]
```diff
@@ -163,19 +170,30 @@ export async function run(nuGetPath: string): Promise<void> {
             }
         }
 
-        // Setting creds in the temp NuGet.config if needed
-        await nuGetConfigHelper.setAuthForSourcesInTempNuGetConfigAsync();
+        tl.debug('nuGetConfigPath'); // satus debugging
```
Comment (satbai) [inline Tasks/NuGetCommandV2/nugetrestore.ts]: remove

File: Tasks/Common/nuget-task-common/NuGetToolRunner2.ts [hunk 6]
```diff
@@ -63,7 +71,12 @@ function prepareNuGetExeEnvironment(
 
     if (authInfo && authInfo.internalAuthInfo) {
         env["VSS_NUGET_ACCESSTOKEN"] = authInfo.internalAuthInfo.accessToken;
-        env["VSS_NUGET_URI_PREFIXES"] = authInfo.internalAuthInfo.uriPrefixes.join(";");
+        env["VSS_NUGET_URI_PREFIXES"] = authInfo.internalAuthInfo.uriPrefixes.join(";")
+    }
+
+    let externalCredentials = buildCredentialJson(authInfo);
```
Comment (satbai) [inline Tasks/Common/nuget-task-common/NuGetToolRunner2.ts]: move this check inside the V2 cred provider check

File: Tasks/NuGetCommandV2/nugetrestore.ts [hunk 7]
```diff
@@ -163,19 +170,30 @@ export async function run(nuGetPath: string): Promise<void> {
...
-
-        if (useConfigFile)
-        {
-            credCleanup = () => tl.rmRF(nuGetConfigHelper.tempNugetConfigPath);
+        if (!useV2CredProvider && !configFile) {
+            // Setting creds in the temp NuGet.config if needed
+            await nuGetConfigHelper.setAuthForSourcesInTempNuGetConfigAsync();
+            tl.debug('Setting nuget.config auth');
+        } else {
+            tl.debug('Not setting temp nuget.config auth');
```
Comment (aldoms) [inline Tasks/NuGetCommandV2/nugetrestore.ts]: nit: add a message saying the new provider will handle auth, or something similar. So we know why we're not adding it.

**Pull request merged**